piet-cairo/src/text.rs

```diff
@@ -220,6 +220,9 @@ impl TextLayout for CairoTextLayout {
 
         let mut htp = hit_test_line_point(&self.font, line, point);
         htp.idx += lm.start_offset;
+        if htp.idx == lm.end_offset && util::has_newline_eof(line) {
+            htp.idx -= 1;
```

Comment:
Same "Note:" comment as below. I think the code is fine, just want a marker in place if and when we do the other Unicode separators.

Reply:
I could make `has_newline_eof` be something like `offset_to_cursor_at_eol` or similar, and it could return a `usize`...

Reply:
This is now... `fn trailing_nlf(s: &str) -> Option<usize>` 🤢